### ISSUE TITLE
Optimizing a script that contains a requirejs call will place amdefine call dependencies into the optimized module instead of requirejs.define calls.

### ISSUE THREAD

**Issue by restarian** [requirejs/requirejs#1702]
Optimizing modules which contain a requirejs call (*like below*), the optimizer will place the dependencies of the requirejs call as *define()* calls into the optimized bundle. however, shouldn't these be inserted as *requirejs.define()* calls? 

**entry.js**
```  
  var requirejs = require("requirejs")
  requirejs(["./second_module"], function(here, two) {
    return {id: "entry"}
  })
```

**second_module.js**
```
  if (typeof define !== 'function') { var define = require('amdefine')(module); }
   define("second_module", [], function() {
       return {id: "second"}
  })
```

**r.js returns this as an optimized  module**
```
  define("second_module", [], function() {
      return {
          id: "second"
      };
  });
  var requirejs = require("requirejs");
   requirejs([ "./second_module" ], function(second) {
      return {
          id: "entry"
      };
  });
  define("entry", function() {});
```

**This is what the optimized module should look like:**
```
  requirejs.define("second_module", [], function() {
      return {
          id: "second"
      };
  });
  requirejs([ "./second_module" ], function(second) {
      return {
          id: "entry"
      };
  });
```

**Comment by restarian** [comment]
Also, I believe that is what this issue is about -> https://github.com/jrburke/amdefine/issues/4

**Comment by jrburke** [comment]
The bundler purposely outputs `define()` calls, as module definitions are just standard AMD. They should work across loaders. Loader options and loader calls to start module loading are specific to a loader, so the `requirejs()` calls are used for those calls.

The amdefine issue mentioned is around how to resolve module IDs, and not specific to module definitions, which is what `define()` provides. How a loader decides to resolve the string IDs mentioned in a `define()` call are loader specific. Looking at that amdefine issue now, I do not see it working out to use a different module resolution than what is in node. It gets hard to bridge the module registries of requirejs and node in those cases.

Closing as working as designed.

**Comment by restarian** [comment]
Well, you never fully grasped what that person was saying. However, setting *that* issue aside, the AMD specification is defied by requirejs as it uses two different loaders within the same platform. Amdefine is the work around to that requirejs specification deviation when in nodejs. At the very least, requirejs should not put those superfluous define calls into the module optimizer like it is doing above. Ideally, it needs to use the requirejs.define syntax. 

The bottom line is: it may not be cost effective to put that much of a polish on it.
